packages/python-packages/apiview-stub-generator/apistub/_stub_generator.py

```diff
@@ -121,16 +122,52 @@ def _parse_arg(self, name):
                 value = None
         return value
 
+    def get_extras_require_from_wheel(self):
+        # Locate the .dist-info directory
+        temp_dir = Path(self.wheel_path)
+        dist_info_dirs = list(temp_dir.glob("*.dist-info"))
+        if not dist_info_dirs:
+            raise ValueError("No .dist-info directory found in the wheel.")
+
+        dist_info_dir = dist_info_dirs[0]
+
+        # Use PathDistribution to load metadata
+        dist = PathDistribution(dist_info_dir)
+        extras_require = dist.metadata.get_all("Provides-Extra")
+
+        # Clean up the temporary directory
+        shutil.rmtree(temp_dir)
+
+        return extras_require or []
+
+    def install_extra_dependencies(self, extras_require):
+        for extra in extras_require:
+            if ':' in extra:
+                logging.info(f"Skipping conditional extra dependency: {extra}")
+                continue
+            logging.info(f"Installing extra dependency: {extra}")
+            try:
+                check_call([sys.executable, "-m", "pip", "install", f"{self.pkg_path}[{extra}]", "-q"])
+            except:
```

Comment:
Thoughts on this @tjprescott ? 

I thought this would be a great way to bypass the "DO we need extras though?", as it will be a fairly uncommon occurrence.

Reply:
After chatting with @swathipil this makes sense to me.